### PR TITLE
add new vep benchmark

### PR DESCRIPTION
**Pull request by naliboff**
This PR adds a new benchmark from Gerya 2019 (Introduction to Numerical Geodynamic Modeling), which examines shortening of a visco plastic or viscoelastic plastic block under compression. This benchmarking will provide a unique point of comparison for #2532 and #2314, and joint work with @alarshi, @cedrict, @jdannberg, @bobmyhill, and others. 

A number of questions and decisions before proceeding further:

1. Should we provide a unique folder for each set of benchmarks in this folder? I think the answer is likely yes if we add any additional files?
2. Should we a README.md file to the main folder and any sub-folders? There is already quite a bit of documentation in the prm files, but this would be useful if we add additional files to sub-folders.
3. The points value post processor unfortunately does not output strain rate. Should I add a python script that reads in VTU data and then plots a strain rate profile?

* [x] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).
* [X] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/master/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/master/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

**Comment by bangerth** [comment]
1. Yes.
2. Yes, a short README.md always makes sense.
3. That would actually be quite a neat idea -- it would be a nice opportunity to introduce another way of postprocessing information.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019_vp.prm
Comment (bangerth) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019_vp.prm:49]:
```suggestion
    set Max pre-Newton nonlinear iterations      = 500 # Ensures only defect-correction Picard method is used
```

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019_vp.prm
Comment (bangerth) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019_vp.prm:74]:
```suggestion
# Globally refine the mesh. A minimum of 3 global refinements
```

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019_vp.prm
Comment (bangerth) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019_vp.prm:97]:
```suggestion
    set Function expression = if (x<50e3, vel, -vel); if (y<50e3, -vel, vel);
```

File: benchmarks/viscoelastic_plastic_shear_bands/README.md
Comment (tjhei) [inline benchmarks/viscoelastic_plastic_shear_bands/README.md:1]:
```suggestion
The folders in this directory contain multiple benchmarks for examining the 
```

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/README.md
Comment (tjhei) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/README.md:16]:
```suggestion
include viscoelasticity (i.e., viscoelastic-plastic rheology).
```

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm
Comment (bobmyhill) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm:11]:
```suggestion
# plastic (here, visco-plastic) block, which contains a weak 
```
(for consistency)

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm
Comment (bobmyhill) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm:49]: explain why?

**Review by bobmyhill — CHANGES_REQUESTED** [review]

**Comment by naliboff** [comment]
@bangerth @tjhei @bobmyhill - Thanks for the preliminary reviews! All of the necessary files, including a python script, are now present and this should be ready for a second review.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py
Comment (tjhei) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py:4]: This is in ParaView or a standalone python code? Why don't you say ``python gerya_2019_analysis.py``?

Comment (naliboff) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py]: That would work as well. The command I listed is for when you want to execute the script while working inside ipython (or similar). I find this helpful for doing quick debugging and testing without needing to rerun the whole script. In any case, good to spell out both options, which is now included in the documentation. In the last commit I also specified the versions of each package in the documentation.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py
Comment (tjhei) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py:5]:
```suggestion
#   python gerya_2019_analysis.py
```

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py
Comment (tjhei) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py:6]:
```suggestion
# Within a python interpreter (e.g., a Jupyter notebook), 
``j

Comment (tjhei) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py:6]: ipython? You are living in the past! :-)

**Comment by naliboff** [comment]
@tjhei - you have no idea, it is VI on side of the screen and ipython on the other :-)

I've addressed the remaining comments, but did notice that two of the testers are failing. I thinking this is due to moving files around associated with the tests and cookbooks, but it is not occurring for the no-unity build case. Aside from resolving this, any additional requests for changes?

**Comment by tjhei** [comment]
> I thinking this is due to moving files around associated with the tests and cookbooks, but it is not occurring for the no-unity build case. Aside from resolving this, any additional requests for changes?

The error is
```
File '*.prm' missing in /jenkins/workspace/aspect_PR-3825/benchmarks/viscoelastic_plastic_shear_bands.
```
Can you adjust https://github.com/geodynamics/aspect/blob/master/benchmarks/check.mk by adding a section for viscoelastic_plastic_shear_bands? Let me know if you need helpl,.

**Comment by tjhei** [comment]
Something like:
``
viscoelastic_plastic_shear_bands/: dummy
	@$(def); run_all_prms $@/gerya_2019
	@$(def); run_all_prms $@/anotherfolder
```

**Comment by tjhei** [comment]
did you figure out what to do, @naliboff ?

**Comment by naliboff** [comment]
@tjhei - Sorry for leaving this hanging for so long and hopefully I've added the right changes to check.mk. I also added two commits that enable using material averaging with viscoelasticity (viscosity averaging only options) and an associated test.

**Comment by tjhei** [comment]
this looks good, but you need to update the test results (screen output changed on master)

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
I am happy. @jdannberg ?

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py
Comment (jdannberg) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py:73]: more than one value?

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py
Comment (jdannberg) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py:74]: I thought that was due to the general format of how we write the output (we write out unstructured meshes, and so we have to give the locations of the corners of each cell). Bt maybe you don't mean what I think you mean here.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm
Comment (jdannberg) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm:7]: You don't need this, correct? (it's already in the other file you include)

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm
Comment (jdannberg) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm:47]: The five parameters above are the same as in the original file, so you don't need them here. 
(unless you think it's important to list them as a reminder for the user)

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm
Comment (jdannberg) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm:89]: This is just a thought for next time (you don't actually have to change anything). But if you want to prescribe the velocity in m/s, why didn't you just set Use years in output instead of seconds to false? The strain rate output won't change.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm
Comment (jdannberg) [inline benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm:119]: I don't think you need this whole section if you don't actually have any boundary conditions for the composition.

**Review by jdannberg — COMMENTED** [review]
This looks good to me! I've made a few suggestions, but I don't want to hold up the merging of this any longer, so feel free to ignore them (they are only minor points) or merge after you've fixed them. 
Sorry this took me so long, and @naliboff thank you for contributing this!

**Review by jdannberg — APPROVED** [review]

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm
Comment (naliboff) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm:119]: Yep, this could be deleted. However, I like to keep this section in as a reminder that the compositional fields in fact have no boundary conditions. Purely there to avoid confusion (on my end) later on :)

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm
Comment (naliboff) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vp.prm:89]: Yep, removed!

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm
Comment (naliboff) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm]: Good catch, no reason to keep these in.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py
Comment (naliboff) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py]: Yes, when I looked at the raw data you can see the same coordinate listed more than once and having a different value. This may be due to the reason you specified below. In any case, I remove the associated text to avoid confusion.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm
Comment (naliboff) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_vep.prm]: yep, removed.

File: benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py
Comment (naliboff) [reply benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py]: You are probably correct that vertices are repeated even without the interpolate output option. Regardless, in the raw VTU data specific points (vertices) are list more than once, and as you stated this is almost certainly due to the way we write out unstructured data. Interpolating the output in ASPECT should then just cause further repetition of some vertices. In any case, as noted above, I removed this comment to avoid confusion :)

**Comment by naliboff** [comment]
@jdannberg - Thanks for the review and catches! I addressed the comments in the last commit, and hopefully nothing changes in the tester. Once this is merged, I'll rebase the plastic damper PR and get that ready for review.

**Comment by naliboff** [comment]
@jdannberg - Ok to merge with the last commit?

**Pull request merged**
